### ISSUE TITLE
Mac OS X Shaders issues

### ISSUE THREAD

**Issue by RedwanFox** [SpringCabal/LD34#16]
Trying to play on Macbook Pro Retina early 2013 with nvidia gpu with Mac OS X 10.11.2. Main base and mushrooms are black. Spring engine doesn't generate any log. 

![Screenshot](http://i.imgur.com/KFxYsLH.jpg)

**Comment by Anarchid** [comment]
A hack to obtain the log:

```
cd Flove.app
./Contents/MacOS/spring > ~/flove.log
```

I'm not sure it will actually save the log in flove.log but at least you'll probably be able to see the log output in the CLI and grep it for renderer.

I'd suspect this is it failing to use the nvidia gpu: i get the same on a mac which doesn't have a discrete card.

**Comment by RedwanFox** [comment]
It seems Spring used NVIDIA GPU. Also, only central structures on map are broken. Trees on the map borders and newly grown flowers are fine. 

Grep results:

```
MacBook-Pro-Redwan:~ redwan$ grep -e "[R|r]end" flove.log
AllowDeferredMapRendering = 1
AllowDeferredModelRendering = 1
Available log sections: KeyBindings, AutohostInterface, GameServer, Net, Font, VFS, Path, CSMFGroundTextures, RoamMeshDrawer, BumpWater, DynWater, SkyBox, DecalsDrawerGL4, FarTextureHandler, Shader, Piece, Model, ModelDrawer, OBJParser, WorldObjectModelRenderer, Texture, CregSerializer, ArchiveScanner, CrashHandler, Sound, LuaSocket, GroundMoveType, UnitScript
GL renderer:  NVIDIA GeForce GT 650M OpenGL Engine
[f=0000000] [Sound]   Renderer:       Software
[f=0000000] Switching to ROAM Mesh Rendering
[f=0000000] [LuaUI] Loading widget:      Deferred rendering     <gfx_deferred_rendering.lua>
[f=0000000] Removing deferred rendering widget: failed to use GLSL shader
[f=0000808] [KillRendering][1]
MacBook-Pro-Redwan:~ redwan$ grep -e "renderer" flove.log
```

ScreenShot:

![Screenshot](http://i.imgur.com/8EgkiFH.jpg)

**Comment by Anarchid** [comment]
Flowers don't receive any shaders, so that's why they're fine.

Hypothesis: this is an issue specifically with the outline shader?

**Comment by RedwanFox** [comment]
Yup, according to log thats the problem. Most of the log file consists of `failed to use GLSL shader` messages.

**Comment by Anarchid** [comment]
The real question, is it outline, or do all shaders fail to compile?

Try modifying the archive `Content/share/games/spring/games/flove.sd7` by removing files similar to `LuaUI/Widgets/*outline*.lua` from said archive.

**Comment by RedwanFox** [comment]
Removed the only file `gfx_outline.lua` and reconstructed the archive. Game doesn't start now, saying that it can't find `flove 1.1` archive.

**Comment by Anarchid** [comment]
Possible reasons: archive no longer having sd7 suffix, or archive being solid.

**Comment by Anarchid** [comment]
Ah, nevermind. The outline hypothesis was stupid anyway: everything would have been affected by that.

Can you post the full log?

**Comment by gajop** [comment]
From the screenshot it seems that the mushroom and treeshaders are failing: https://github.com/SpringCabal/LD34/tree/master/ModelMaterials as well as the wisp one: https://github.com/SpringCabal/LD34/blob/master/LuaRules/Gadgets/wisp_animation.lua
It may be prudent to raise this with the engine if you could provide an infolog?

Raise @abma @rtri

**Comment by abma** [comment]
> Spring engine doesn't generate any log. 

no, you are looking in the wrong directories. try ~/.config/spring/

without infolog.txt its not really possible to help.

**Comment by RedwanFox** [comment]
Full log has arrived.
[flove.log.txt](https://github.com/SpringCabal/Flove/files/65429/flove.log.txt)

**Comment by abma** [comment]
tons of shader errors in the log file, i guess devs now have something to work now :-)

thanks!

**Comment by Anarchid** [comment]
Looks like mac nvidia drivers are for some reason much more strict than both nvidia linux/windows ones as well as those for ATI!
